sdb/commands/stacks.py

```diff
@@ -0,0 +1,255 @@
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+import argparse
+from typing import Iterable, Tuple
+from collections import Counter, defaultdict
+
+import drgn
+from drgn.helpers.linux.list import list_for_each_entry
+from drgn.helpers.linux.pid import for_each_task
+
+import sdb
+
+
+#
+# Note: This is a rudimentary version of what the command could/should be.
+#
+# On the high-level, it could be a `Locator`, or something similar, where
+# objects could be passed as input from the pipeline dispatching different
+# methods depending on the type of object. E.g. input could be a namespace
+# object and we print all the task structs within it, or it could be just
+# a list of task structs passed from a previous command for filtering.
+# Another option would be to decouple the stack listing, filtering, and
+# pretty-printing functionality to independent SDB commands.
+#
+# There are also other lower-level usability improvements like supporting
+# filtering by `function+offset` with the `-c` option, or by namespace ID
+# using `-n <ID>`.
+#
+# Finally, the command lacks any support for userland targets.
+#
+# SDB is still in its early stages and hasn't been used enough for us to
+# be clear which use cases really matter. In the meantime if we don't have
+# anything that provides this functionality it won't be easy to do this
+# exploration. The version below is a good enough for the time being
+# providing some basic functionality and being our tracer bullet for
+# future iterations.
+#
+class Stacks(sdb.Command):
+    # pylint: disable=too-few-public-methods
+
+    names = ["stacks"]
+
+    def _init_argparse(self, parser: argparse.ArgumentParser) -> None:
+        parser.add_argument(
+            '-a',
+            '--all',
+            action='store_true',
+            help='Print all grouped threads, instead of just a count')
```

Comment:
I'll change this to:
```
list all threads for each unique stack trace instead of printing a single representative thread
```
I think it is better to start with a verb when possible

Reply:
Sounds good to me.